packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -600,7 +637,13 @@ describe('TransactionController', () => {
         }
       });
 
-    const mockFindNetworkClientIdByChainId = jest.fn();
+    ({ messenger: messengerMock, approve: approveTransaction } =
+      buildMockMessenger({
+        addRequest: addRequestMockOptions,
+        getNetworkClientById: mockGetNetworkClientById,
+        findNetworkClientIdByChainId: jest.fn(),
+      }));
```

Comment:
I know you didn't introduce this pattern but I find it so confusing that we're assigning the assigning the globally scoped mock this way...

Reply:
yeah... i agree. `newController` needs to be updated to return messenger and this approve callback